packages/astro/src/vite-plugin-astro-server/index.ts

```diff
@@ -423,7 +424,21 @@ async function handleRoute(
 			if (computedMimeType) {
 				contentType = computedMimeType;
 			}
-			res.writeHead(200, { 'Content-Type': `${contentType};charset=utf-8` });
+
+      const header: any = {
+        'Content-Type': `${contentType};charset=utf-8`
+      };
+
+			let cookieHeader = [];
+			for (let cok of result.cookies.headers()) {
+				cookieHeader.push(cok);
+			}
+
+      if (cookieHeader.length > 0) {
+        header['Set-Cookie'] = cookieHeader;
+      }
```

Comment:
This is already being done here: https://github.com/withastro/astro/blob/a5e3ecc8039c1e115ce5597362e18cd35d04e40b/packages/astro/src/vite-plugin-astro-server/index.ts#L67-L70

I think we should keep it the old way because it's the same way that adapters grab the cookies and we should have dev-prod symmetry. There is no result object in adapters.

Reply:
Cool, I converted it to
- create a response object (from the simple result)
- attach the cookie to the response
- call `writeWebResponse`

is that what you were looking for, or should i just have replaced the pure cookie setting?

Reply:
You don't need to do the second step:

> * attach the cookie to the response

because it's already done here: https://github.com/AirBorne04/astro/blob/5443c70df50772c617e8c106f0dc603489476035/packages/astro/src/vite-plugin-astro-server/index.ts#L67-L71

Reply:
The `getSetCookiesFromResponse` grabs the cookies. We store then on the actual `Response` object as a symbol.

Reply:
sorry, i should have added a pointer to the code changes.

> We store them on the actual `Response` object as a symbol.

that is what i was referring to, i am calling `attachToResponse` so that the `getSetCookiesFromResponse` can use the cookie data, since i am creating that `Response` only for the `dev` server.

Here is the corresponding [code change](https://github.com/withastro/astro/pull/5060/files#r995572557)